### PR TITLE
Ibmcloud Network perf test

### PR DESCRIPTION
**Pull request by SachinNinganure**
/hold

### CODE REVIEW

**Comment by SachinNinganure** [comment]
/pj-rehearse pull-ci-openshift-qe-ocp-qe-perfscale-ci-main-ibmcloud-4.18-nightly-x86-data-path-9nodes

**Comment by SachinNinganure** [comment]
/pj-rehearse pull-ci-openshift-qe-ocp-qe-perfscale-ci-main-ibmcloud-4.18-nightly-x86-data-path-9nodes

**Comment by SachinNinganure** [comment]
/pj-rehearse pull-ci-openshift-qe-ocp-qe-perfscale-ci-main-ibmcloud-4.18-nightly-x86-data-path-9nodes

**Comment by SachinNinganure** [comment]
/pj-rehearse pull-ci-openshift-qe-ocp-qe-perfscale-ci-main-ibmcloud-4.18-nightly-x86-data-path-9nodes

**Comment by SachinNinganure** [comment]
/pj-rehearse pull-ci-openshift-qe-ocp-qe-perfscale-ci-main-ibmcloud-4.18-nightly-x86-data-path-9nodes

**Comment by SachinNinganure** [comment]
/pj-rehearse pull-ci-openshift-qe-ocp-qe-perfscale-ci-main-ibmcloud-4.18-nightly-x86-data-path-9nodes

**Comment by SachinNinganure** [comment]
/pj-rehearse pull-ci-openshift-qe-ocp-qe-perfscale-ci-main-ibmcloud-4.18-nightly-x86-data-path-9nodes

**Comment by SachinNinganure** [comment]
/pj-rehearse pull-ci-openshift-qe-ocp-qe-perfscale-ci-main-ibmcloud-4.18-nightly-x86-data-path-9nodes

**Comment by SachinNinganure** [comment]
/pj-rehearse pull-ci-openshift-qe-ocp-qe-perfscale-ci-main-ibmcloud-4.18-nightly-x86-data-path-9nodes

**Comment by SachinNinganure** [comment]
/pj-rehearse pull-ci-openshift-qe-ocp-qe-perfscale-ci-main-ibmcloud-4.18-nightly-x86-data-path-9nodes

**Comment by SachinNinganure** [comment]
/pj-rehearse pull-ci-openshift-qe-ocp-qe-perfscale-ci-main-ibmcloud-4.18-nightly-x86-data-path-9nodes

File: ci-operator/step-registry/openshift-qe/perfscale-ibm-data-path-sg/openshift-qe-perfscale-ibm-data-path-sg-commands.sh
Comment (liqcui) [inline ci-operator/step-registry/openshift-qe/perfscale-ibm-data-path-sg/openshift-qe-perfscale-ibm-data-path-sg-commands.sh:14]: @SachinNinganure where you get the value of region?

**Comment by SachinNinganure** [comment]
/pj-rehearse pull-ci-openshift-qe-ocp-qe-perfscale-ci-main-ibmcloud-4.18-nightly-x86-data-path-9nodes

File: ci-operator/step-registry/openshift-qe/perfscale-ibm-data-path-sg/openshift-qe-perfscale-ibm-data-path-sg-commands.sh
Comment (MayXuQQ) [inline ci-operator/step-registry/openshift-qe/perfscale-ibm-data-path-sg/openshift-qe-perfscale-ibm-data-path-sg-commands.sh:20]: ibmcloud resource group $resource_group_name || exit 1

**Comment by SachinNinganure** [comment]
/pj-rehearse pull-ci-openshift-qe-ocp-qe-perfscale-ci-main-ibmcloud-4.18-nightly-x86-data-path-9nodes

**Comment by SachinNinganure** [comment]
/pj-rehearse pull-ci-openshift-qe-ocp-qe-perfscale-ci-main-ibmcloud-4.18-nightly-x86-data-path-9nodes

**Comment by SachinNinganure** [comment]
/pj-rehearse pull-ci-openshift-qe-ocp-qe-perfscale-ci-main-ibmcloud-4.18-nightly-x86-data-path-9nodes

**Comment by SachinNinganure** [comment]
/pj-rehearse pull-ci-openshift-qe-ocp-qe-perfscale-ci-main-ibmcloud-4.18-nightly-x86-data-path-9nodes

**Comment by SachinNinganure** [comment]
/pj-rehearse pull-ci-openshift-qe-ocp-qe-perfscale-ci-main-ibmcloud-4.18-nightly-x86-data-path-9nodes

**Comment by SachinNinganure** [comment]
/pj-rehearse pull-ci-openshift-qe-ocp-qe-perfscale-ci-main-ibmcloud-4.18-nightly-x86-data-path-9nodes

**Comment by SachinNinganure** [comment]
/pj-rehearse pull-ci-openshift-qe-ocp-qe-perfscale-ci-main-ibmcloud-4.18-nightly-x86-data-path-9nodes

File: ci-operator/config/openshift-qe/ocp-qe-perfscale-ci/openshift-qe-ocp-qe-perfscale-ci-main__ibmcloud-4.18-nightly-x86.yaml
Comment (MayXuQQ) [inline ci-operator/config/openshift-qe/ocp-qe-perfscale-ci/openshift-qe-ocp-qe-perfscale-ci-main__ibmcloud-4.18-nightly-x86.yaml:85]: need change the order?

Comment (SachinNinganure) [reply ci-operator/config/openshift-qe/ocp-qe-perfscale-ci/openshift-qe-ocp-qe-perfscale-ci-main__ibmcloud-4.18-nightly-x86.yaml:85]: but openshift-qe-perfscale-ibm-data-path-sg will fail and I don't want to run network tests before checking if the rules are added.... but the cucushift is failing with imagepullbackoff

**Comment by SachinNinganure** [comment]
/pj-rehearse pull-ci-openshift-qe-ocp-qe-perfscale-ci-main-ibmcloud-4.18-nightly-x86-data-path-9nodes

File: ci-operator/config/openshift-qe/ocp-qe-perfscale-ci/openshift-qe-ocp-qe-perfscale-ci-main__ibmcloud-4.18-nightly-x86.yaml
Comment (SachinNinganure) [reply ci-operator/config/openshift-qe/ocp-qe-perfscale-ci/openshift-qe-ocp-qe-perfscale-ci-main__ibmcloud-4.18-nightly-x86.yaml:84]: The code is working fine I could see the rules added to the --wide security group. I have taken off the debug "sleep" code off now.
currently it failed for (runtime error: invalid memory address or nil pointer dereference)}

@liqcui  suggested to increase timeout to 8 hours. Please review

**Comment by SachinNinganure** [comment]
/pj-rehearse pull-ci-openshift-qe-ocp-qe-perfscale-ci-main-ibmcloud-4.18-nightly-x86-data-path-9nodes

File: ci-operator/config/openshift-qe/ocp-qe-perfscale-ci/openshift-qe-ocp-qe-perfscale-ci-main__ibmcloud-4.18-nightly-x86.yaml
Comment (MayXuQQ) [reply ci-operator/config/openshift-qe/ocp-qe-perfscale-ci/openshift-qe-ocp-qe-perfscale-ci-main__ibmcloud-4.18-nightly-x86.yaml:84]: you can add the cucushift-installer-wait step, and set SLEEP_DURATION <20h
when stop debugging, delete the cucushift-installer-wait pod, the post steps will be run

**Comment by SachinNinganure** [comment]
/pj-rehearse pull-ci-openshift-qe-ocp-qe-perfscale-ci-main-ibmcloud-4.18-nightly-x86-data-path-9nodes

**Comment by SachinNinganure** [comment]
/pj-rehearse pull-ci-openshift-qe-ocp-qe-perfscale-ci-main-ibmcloud-4.18-nightly-x86-data-path-9nodes

**Comment by SachinNinganure** [comment]
@krishvoor @liqcui

**Comment by liqcui** [comment]
looks good to me, wait for other's review comments

File: ci-operator/step-registry/openshift-qe/perfscale-ibm-data-path-sg/openshift-qe-perfscale-ibm-data-path-sg-commands.sh
Comment (krishvoor) [reply ci-operator/step-registry/openshift-qe/perfscale-ibm-data-path-sg/openshift-qe-perfscale-ibm-data-path-sg-commands.sh:24]: s/ibmcloud/${IBMCLOUD_CLI}/g

Please replace references of `ibmcloud` with ${IBMCLOUD_CLI}

File: ci-operator/config/openshift-qe/ocp-qe-perfscale-ci/openshift-qe-ocp-qe-perfscale-ci-main__ibmcloud-4.18-nightly-x86.yaml
Comment (krishvoor) [inline ci-operator/config/openshift-qe/ocp-qe-perfscale-ci/openshift-qe-ocp-qe-perfscale-ci-main__ibmcloud-4.18-nightly-x86.yaml:79]: Will this be a periodic job?

**Review by krishvoor — COMMENTED** [review]
Minor nits, other looks good

File: ci-operator/config/openshift-qe/ocp-qe-perfscale-ci/openshift-qe-ocp-qe-perfscale-ci-main__ibmcloud-4.18-nightly-x86.yaml
Comment (krishvoor) [reply ci-operator/config/openshift-qe/ocp-qe-perfscale-ci/openshift-qe-ocp-qe-perfscale-ci-main__ibmcloud-4.18-nightly-x86.yaml:79]: If so, please add the cron job

Comment (SachinNinganure) [reply ci-operator/config/openshift-qe/ocp-qe-perfscale-ci/openshift-qe-ocp-qe-perfscale-ci-main__ibmcloud-4.18-nightly-x86.yaml:79]: @qiliRedHat  Please let know if I will have to make this periodic

File: ci-operator/step-registry/openshift-qe/perfscale-ibm-data-path-sg/openshift-qe-perfscale-ibm-data-path-sg-commands.sh
Comment (qiliRedHat) [inline ci-operator/step-registry/openshift-qe/perfscale-ibm-data-path-sg/openshift-qe-perfscale-ibm-data-path-sg-commands.sh:28]: @SachinNinganure  You have got the sg above in line 25. Is this line just to output for debugging?

File: ci-operator/config/openshift-qe/ocp-qe-perfscale-ci/openshift-qe-ocp-qe-perfscale-ci-main__ibmcloud-4.18-nightly-x86.yaml
Comment (qiliRedHat) [reply ci-operator/config/openshift-qe/ocp-qe-perfscale-ci/openshift-qe-ocp-qe-perfscale-ci-main__ibmcloud-4.18-nightly-x86.yaml:79]: @SachinNinganure No for IBM Cloud we don't run cron now.

Comment (krishvoor) [reply ci-operator/config/openshift-qe/ocp-qe-perfscale-ci/openshift-qe-ocp-qe-perfscale-ci-main__ibmcloud-4.18-nightly-x86.yaml:79]: @qiliRedHat - thank you!
@SachinNinganure: Please add `always_run: false`. For reference [URL](https://github.com/openshift/release/blob/master/ci-operator/config/openshift-qe/ocp-qe-perfscale-ci/openshift-qe-ocp-qe-perfscale-ci-main__aws-4.18-nightly-x86.yaml#L103)

File: ci-operator/step-registry/openshift-qe/perfscale-ibm-data-path-sg/openshift-qe-perfscale-ibm-data-path-sg-commands.sh
Comment (SachinNinganure) [reply ci-operator/step-registry/openshift-qe/perfscale-ibm-data-path-sg/openshift-qe-perfscale-ibm-data-path-sg-commands.sh:28]: line 25 is to fetch the security group with the name cluster-wide (non-default, due to this the code was stuck and long pending to be running) which I will be using to add the rules at line 37 and 38, while line 28 will display the present security groups of the cluster. thank you for the question

File: ci-operator/step-registry/openshift-qe/perfscale-ibm-data-path-sg/openshift-qe-perfscale-ibm-data-path-sg-commands.sh
Comment (SachinNinganure) [reply ci-operator/step-registry/openshift-qe/perfscale-ibm-data-path-sg/openshift-qe-perfscale-ibm-data-path-sg-commands.sh:12]: removed

File: ci-operator/step-registry/openshift-qe/perfscale-ibm-data-path-sg/openshift-qe-perfscale-ibm-data-path-sg-commands.sh
Comment (krishvoor) [inline ci-operator/step-registry/openshift-qe/perfscale-ibm-data-path-sg/openshift-qe-perfscale-ibm-data-path-sg-commands.sh]: Can you update this to
```
${IBMCLOUD_CLI} resource group $resource_group_name || exit 1
${IBMCLOUD_CLI} target -g $resource_group_name -r $region
```

File: ci-operator/step-registry/openshift-qe/perfscale-ibm-data-path-sg/openshift-qe-perfscale-ibm-data-path-sg-commands.sh
Comment (krishvoor) [reply ci-operator/step-registry/openshift-qe/perfscale-ibm-data-path-sg/openshift-qe-perfscale-ibm-data-path-sg-commands.sh]: I still see references to  ```export IBMCLOUD_HOME=/output``` do we need it?

File: ci-operator/step-registry/openshift-qe/perfscale-ibm-data-path-sg/openshift-qe-perfscale-ibm-data-path-sg-commands.sh
Comment (krishvoor) [inline ci-operator/step-registry/openshift-qe/perfscale-ibm-data-path-sg/openshift-qe-perfscale-ibm-data-path-sg-commands.sh]: Update this `ibmcloud --> ${IBMCLOUD_CLI}`

File: ci-operator/step-registry/openshift-qe/perfscale-ibm-data-path-sg/openshift-qe-perfscale-ibm-data-path-sg-commands.sh
Comment (krishvoor) [inline ci-operator/step-registry/openshift-qe/perfscale-ibm-data-path-sg/openshift-qe-perfscale-ibm-data-path-sg-commands.sh]: On side note: please Update this ibmcloud --> ${IBMCLOUD_CLI}

File: ci-operator/step-registry/openshift-qe/perfscale-ibm-data-path-sg/openshift-qe-perfscale-ibm-data-path-sg-commands.sh
Comment (krishvoor) [reply ci-operator/step-registry/openshift-qe/perfscale-ibm-data-path-sg/openshift-qe-perfscale-ibm-data-path-sg-commands.sh]: I see this is resolved in latest commit, thanks

File: ci-operator/step-registry/openshift-qe/perfscale-ibm-data-path-sg/openshift-qe-perfscale-ibm-data-path-sg-commands.sh
Comment (krishvoor) [reply ci-operator/step-registry/openshift-qe/perfscale-ibm-data-path-sg/openshift-qe-perfscale-ibm-data-path-sg-commands.sh]: I see this is resolved in latest commit, thanks

**Comment by SachinNinganure** [comment]
/pj-rehearse pull-ci-openshift-qe-ocp-qe-perfscale-ci-main-ibmcloud-4.18-nightly-x86-data-path-9nodes

**Comment by SachinNinganure** [comment]
/pj-rehearse pull-ci-openshift-qe-ocp-qe-perfscale-ci-main-ibmcloud-4.18-nightly-x86-data-path-9nodes

**Comment by SachinNinganure** [comment]
/pj-rehearse pull-ci-openshift-qe-ocp-qe-perfscale-ci-main-ibmcloud-4.18-nightly-x86-data-path-9nodes

**Comment by SachinNinganure** [comment]
@krishvoor https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/openshift_release/52542/rehearse-52542-pull-ci-openshift-qe-ocp-qe-perfscale-ci-main-ibmcloud-4.18-nightly-x86-data-path-9nodes/1859842638086148096
it fails at some places... does not find "is"

**Comment by SachinNinganure** [comment]
/pj-rehearse pull-ci-openshift-qe-ocp-qe-perfscale-ci-main-ibmcloud-4.18-nightly-x86-data-path-9nodes

**Comment by SachinNinganure** [comment]
/pj-rehearse pull-ci-openshift-qe-ocp-qe-perfscale-ci-main-ibmcloud-4.18-nightly-x86-data-path-9nodes

**Comment by SachinNinganure** [comment]
/pj-rehearse pull-ci-openshift-qe-ocp-qe-perfscale-ci-main-ibmcloud-4.18-nightly-x86-data-path-9nodes

File: ci-operator/step-registry/openshift-qe/perfscale-ibm-data-path-sg/openshift-qe-perfscale-ibm-data-path-sg-commands.sh
Comment (qiliRedHat) [reply ci-operator/step-registry/openshift-qe/perfscale-ibm-data-path-sg/openshift-qe-perfscale-ibm-data-path-sg-commands.sh:27]: @SachinNinganure Would you update this to 'security group'?

**Comment by qiliRedHat** [comment]
@SachinNinganure Please confirm if you have addressed the comment https://github.com/openshift/release/pull/52542#discussion_r1851480616. Otherwise lgtm.

**Comment by SachinNinganure** [comment]
/pj-rehearse pull-ci-openshift-qe-ocp-qe-perfscale-ci-main-ibmcloud-4.18-nightly-x86-data-path-9nodes

**Comment by SachinNinganure** [comment]
/pj-rehearse pull-ci-openshift-qe-ocp-qe-perfscale-ci-main-ibmcloud-4.18-nightly-x86-data-path-9nodes

**Comment by krishvoor** [comment]
/pj-rehearse pull-ci-openshift-qe-ocp-qe-perfscale-ci-main-ibmcloud-4.18-nightly-x86-data-path-9nodes

**Comment by krishvoor** [comment]
While the last few commits are rebase + make updates not changing the script in question. We do not need another run.
Here's the last successful run [URL](https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/openshift_release/52542/rehearse-52542-pull-ci-openshift-qe-ocp-qe-perfscale-ci-main-ibmcloud-4.18-nightly-x86-data-path-9nodes/1863567794222665728)

**Review by krishvoor — APPROVED** [review]
/lgtm

**Review by krishvoor — APPROVED** [review]
/lgtm

**Comment by SachinNinganure** [comment]
Hi @qiliRedHat   Please approve this to be merged. ty

**Comment by qiliRedHat** [comment]
@SachinNinganure You need some one in [ci-operator/step-registry/cucushift/installer/OWNERS](https://github.com/openshift/release/blob/master/ci-operator/step-registry/cucushift/installer/OWNERS)'s approve merge.

**Comment by SachinNinganure** [comment]
@yunjiang29 Ptal

File: ci-operator/step-registry/cucushift/installer/wait/cucushift-installer-wait-commands.sh
Comment (yunjiang29) [inline ci-operator/step-registry/cucushift/installer/wait/cucushift-installer-wait-commands.sh:4]: @SachinNinganure  I think you can simply revert this file so that the PR doesn't need my approval

**Pull request merged**
